flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobMaster.java

```diff
@@ -280,12 +281,7 @@ public JobMaster(
 
 		resourceManagerLeaderRetriever = highAvailabilityServices.getResourceManagerLeaderRetriever();
 
-		this.slotPool = new SlotPool(
-			rpcService,
-			jobGraph.getJobID(),
-			SystemClock.getInstance(),
-			rpcTimeout,
-			jobMasterConfiguration.getSlotIdleTimeout());
+		this.slotPool = checkNotNull(slotPoolFactory).createSlotPool(jobGraph.getJobID());
```

Comment:
It seems like you could also construct the `SlotPool` outside just pass it directly into the constructor. The factory is only used to deliver the `SlotPool`-object and no reference is kept.

Reply:
The reason why I introduced the factory is that we want to create a `SlotPool` for the `JobID` of the given `JobGraph`. That way we enforce that the `SlotPool` is strictly instantiated with the right `JobID`. If we move the instantiation of the `SlotPool` out of the `JobMaster` we would have to rely on the outside code. Therefore, I would like to keep it like it is.